xcode/Subconscious/SubconsciousTests/Tests_App.swift

```diff
@@ -66,8 +62,8 @@ final class Tests_App: XCTestCase {
         )
         XCTAssertEqual(b.state.nicknameFormField.value, "valid", "Nickname form field is set")
         XCTAssertEqual(b.state.nicknameFormField.isValid, true, "Nickname form field is valid")
-        XCTAssertEqual(b.state.nicknameFormFieldValue, "valid", "Nickname form field value set")
-        XCTAssertEqual(b.state.isNicknameFormFieldValid, true, "Nickname form field is valid")
-        XCTAssertEqual(b.state.nickname, "valid", "Nickname is saved, because form field value is valid")
+        
+        let c = AppModel.update(state: model, action: .submitNickname(Petname.Name("valid")!), environment: AppEnvironment())
+        XCTAssertEqual(c.state.nickname, "valid", "Nickname is saved after submitted valid nickname.")
```

Comment:
Updated to capture new behaviour: `state.nickname` won't be update until `submitNickname` is called with a valid name.